extensions/wikia/Chat2/Chat.class.php

```diff
@@ -150,144 +159,177 @@ public static function blockPrivate($username, $dir = 'add', $kickingUser) {
 	 * @param string $reason
 	 * @param string $dir
 	 *
-	 * @return bool
 	 * @throws DBUnexpectedError
 	 * @throws MWException
 	 */
-	public static function banUserDB($cityId, $banUser, $adminUser, $time, $reason, $dir = 'add') {
+	public static function banUserDB( $cityId, $banUser, $adminUser, $time, $reason, $dir = 'add' ) {
+		if ( empty( $banUser ) || empty( $adminUser ) ) {
+			return;
+		}
+
+		if ( wfReadOnly() ) {
+			return;
+		}
+
+		$adminID = $adminUser->getId();
+		$userID = $banUser->getId();
+
+		if ( $dir == 'remove' ) {
+			$timeLabel = $endOn = null;
+
+			self::deleteUserBanFromDB( $cityId, $userID );
+		} else {
+			if ( Chat::getBanInformation( $cityId, $banUser ) !== false ) {
+				$dir = "change";
+			}
+
+			$timeLabel = self::getTimeLabel( $time );
+			$endOn = time() + $time;
+
+			self::addUserBanToDB( $cityId, $userID, $adminID, $endOn, $reason );
+		}
+
 		ChatHelper::info( __METHOD__ . ': Method called', [
 			'cityId' => $cityId,
-			'banUser' => $banUser,
-			'adminUser' => $adminUser,
+			'banUser' => $userID,
```

Comment:
Its not really defined anywhere and both are used (e.g. ```$wgExternalDatawareDB``` below).  I guess my personal preference is to leave abbreviations and acronyms uppercase if they end a word, but down-case if they're in the middle so they aren't confused with the next hump in the name.